package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -95,10 +134,591 @@ def __init__(self, filename, **kwargs):
+            atom_mapper[atom.ix] = index
+
+        try:
+            ag.bonds
+        except NoDataError:
+            warnings.warn(
+                "No `bonds` attribute in this AtomGroup. Guessing bonds based "
+                "on atoms coordinates")
+            ag.guess_bonds()
+
+        terminal_atom_indices = []
+        for bond in ag.bonds:
+            try:
+                bond_indices = [atom_mapper[i] for i in bond.indices]
+            except KeyError:
+                # one of the atoms of the bond is not part of the atomgroup.
+                # can happen for terminal atoms.
+                # save the bond atom that is in the atomgroup for later
+                terminal_atom_indices.extend([atom_mapper[i]
+                                              for i in bond.indices
+                                              if i in atom_mapper.keys()])
+                # skip adding this bond
+                continue
+            bond_type = RDBONDORDER.get(bond.order, Chem.BondType.SINGLE)
+            mol.AddBond(*bond_indices, bond_type)
+
+        mol.UpdatePropertyCache(strict=False)
+
+        # infer bond orders and formal charges from the connectivity
+        _infer_bo_and_charges(mol, terminal_atom_indices)
+        mol = _standardize_patterns(mol)
+
+        # sanitize
+        Chem.SanitizeMol(mol)
+
+        return mol
+
+
+def _add_mda_attr_to_rdkit(attr, value, mi):
+    """Converts an MDAnalysis atom attribute into the RDKit equivalent and
+    stores it into an RDKit AtomPDBResidueInfo object.
+
+    Parameters
+    ----------
+
+    attr : str
+        Name of the atom attribute in MDAnalysis in the singular form
+    value : object, np.int or np.float
+        Attribute value as found in the AtomGroup
+    mi : rdkit.Chem.rdchem.AtomPDBResidueInfo
+        MonomerInfo object that will store the relevant atom attributes
+    """
+    if isinstance(value, np.generic):
+        # convert numpy types to python standard types
+        value = value.item()
+    if attr == "names":
+        # RDKit needs the name to be properly formatted for a
+        # PDB file (1 letter elements start at col 14)
+        name = re.findall(r'(\D+|\d+)', value)
+        if len(name) == 2:
+            symbol, number = name
+        else:
+            symbol, number = name[0], ""
+        value = "{:>2.2}{:<2.2}".format(symbol, number)
+    # set attribute value in RDKit MonomerInfo
+    rdattr = RDATTRIBUTES[attr]
+    getattr(mi, "Set%s" % rdattr)(value)
+
+
+def _set_atom_property(atom, attr, value):
+    """Saves any attribute and value into an RDKit atom property"""
+    if isinstance(value, (float, np.float)):
+        atom.SetDoubleProp(attr, float(value))
+    elif isinstance(value, (int, np.int)):
+        atom.SetIntProp(attr, int(value))
+    else:
+        atom.SetProp(attr, value)
+
+
+def _infer_bo_and_charges(mol, terminal_atom_indices=[]):
+    """Infer bond orders and formal charges from a molecule.
```

Comment:
I've not tried generating the docs locally, so might be wrong, but I think by default because these are private functions they won't be added to the docs, however I do think that they should be available for users to see. It should be able to force sphinx to build the docs for these if it's not happening already.

(note mainly referring to this and the standardize patterns below)

Reply:
One can tell sphinx to also show undocumented members, or, include it explicitly in the list of members. See as example how `_reopen()` was documented https://github.com/MDAnalysis/mdanalysis/blob/5c0eb207d2141bb4166112fe3af4fe61d0fad2fc/package/MDAnalysis/coordinates/H5MD.py#L181